### PR TITLE
Searchable dropdown fix

### LINKED ISSUE #43: searchable-dropdown should select when pressed enter
Currently it has a problem, it doesn't select the focused dropdown item.

Also, if there is a single options, `enter` should select it even if it is not on focus or it can always focus the first item which will directly support this functionality when the above bug is resolved.

### LINKED ISSUE #44: x_editor_enum_shortcut_map should be documented on the README

### PR DESCRIPTION
**Pull request by harunurhan**
### refactor and fix searchable-dropdown

* Changes from prefix search to ignored case expression search which tries to
match given input in anywhere of possible items.

* Fixes #43

### add documentation for x_editor_enum_shortcut_map

* Fixes #44

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -123,6 +123,19 @@ number
 default: 0
 ```
 
+#### x_editor_enum_shortcut_map
+
+Shortcut map for enum fields, to map shorthand values to enum values
+
+```
+{
+  a: "An value in enum array",
```
Comment (jmartinm) [inline README.md]: This is a `shortcut` and not value in enum array?

File: README.md [hunk 2]
```diff
@@ -123,6 +123,19 @@ number
...
+  abc: "Another value in enum array"
+}
+```
+
+So when `a` typed in enum field and enter is pressed, enum field value will be set to `An value in enum array`.
```
Comment (jmartinm) [inline README.md]: The enum field value will be `Another value in enum array` ?

File: src/shared/pipes/filter-by-expression.pipe.spec.ts [hunk 3]
```diff
@@ -20,17 +20,17 @@
...
 
   beforeEach(() => {
-    pipe = new FilterByPrefixPipe();
+    pipe = new FilterByExpressionPipe();
   });
 
-  it('should filter by prefix', () => {
-    let values = ['ab1', 'ab2', 'ac1', 'bc1'];
+  it('should filter by expression', () => {
+    let values = ['ab1', 'ab2', 'xc1', 'zc1'];
```
Comment (jmartinm) [inline src/shared/pipes/filter-by-expression.pipe.spec.ts]: Would be good to test that your regexp filtering works, for example typing c1 should return 'xc1', 'zc1'

File: README.md (hunk 1 above)
Comment (harunurhan) [inline README.md]: key is `shortcut`, value should be one of the values in enum array.

**Pull request merged**